src/main/java/seedu/address/model/moduleclass/UniqueModuleClassList.java

```diff
@@ -0,0 +1,156 @@
+package seedu.address.model.moduleclass;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.moduleclass.exceptions.DuplicateModuleClassException;
+import seedu.address.model.moduleclass.exceptions.ModuleClassNotFoundException;
+
+/**
+ * A list of {@code ModuleClass} that enforces uniqueness between its elements and does not allow nulls.
+ * A {@code ModuleClass} is considered unique by comparing using {@code ModuleClass#isSameModuleClass(ModuleClass)}.
+ * As such, adding and updating of {@code ModuleClass} uses ModuleClass#isSameModuleClass(ModuleClass) for equality
+ * so as to ensure that the {@code ModuleClass} being added or updated is unique in terms of identity in the
+ * UniqueModuleClassList. However, the removal of a {@code ModuleClass} uses ModuleClass#equals(Object) so as to ensure
+ * that the {@code ModuleClass} with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see ModuleClass#isSameModuleClass(ModuleClass)
+ */
+public class UniqueModuleClassList implements Iterable<ModuleClass> {
+
+    private final ObservableList<ModuleClass> internalList = FXCollections.observableArrayList();
+    private final ObservableList<ModuleClass> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent {@code ModuleClass} as the given argument.
+     *
+     * @throws NullPointerException if the given argument is null.
+     */
+    public boolean contains(ModuleClass toCheck) throws NullPointerException {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameModuleClass);
+    }
+
+    /**
+     * Adds a {@code ModuleClass} to the list.
+     * The {@code ModuleClass} must not already exist in the list.
```

Comment:
``` The {@code ModuleClass} must not exist in the current list ```

Reply:
The JavaDoc was adapted from an equivalent JavaDoc in `UniqueStudentList`.

I would be more inclined to retain both in their current expressions.

Reply:
Ok lets retain it in that case